packages/toolbar/src/browser/toolbar.tsx

```diff
@@ -143,7 +143,7 @@ export class ToolbarImpl extends TabBarToolbar {
 
     protected renderGroupsInColumn(groups: ToolbarItem[][], alignment: ToolbarAlignment): React.ReactNode[] {
         const nodes: React.ReactNode[] = [];
-        groups.forEach((group, groupIndex) => {
+        groups?.forEach((group, groupIndex) => {
```

Comment:
Suggestion: I don't think that this is the correct fix. Looking at the method definition, we shouldn't even be allowed to pass `undefined` in here. The caller site of this method should take care of not passing in `undefined`.